### PR TITLE
Fix bug in Dirichlet.rsample()

### PR DESCRIPTION
**Pull request by fritzo**
This adds more tests for `Beta.rsample()` and `Dirichlet.rsample()`.

### CODE REVIEW

**Comment by fritzo** [comment]
cc @martinjankowiak

File: torch/distributions/utils.py [hunk 1]
```diff
@@ -1,9 +1,21 @@
+from collections import namedtuple
 from functools import update_wrapper
 from numbers import Number
 
 import torch
-from torch.autograd import Variable
 import torch.nn.functional as F
+from torch.autograd import Variable
+
+# This follows semantics of numpy.finfo.
```
Comment (fritzo) [inline torch/distributions/utils.py]: @alicanb Here's a first draft of a numerics table.

File: test/test_distributions.py [hunk 2]
```diff
@@ -958,6 +872,217 @@ def test_invalid_parameter_broadcasting(self):
...
+            den = (1.0 + alpha)**2 * (1.0 + 2.0 * alpha)**3
+            expected_grad = (num / den).data
+            self.assertEqual(actual_grad, expected_grad, 0.002, '\n'.join([
+                "alpha = alpha_c + %.2g" % shift,
+                "expected_grad: %.5g" % expected_grad,
+                "actual_grad: %.5g" % actual_grad,
+                "error = %.2g" % torch.abs(expected_grad - actual_grad).max(),
+            ]))
+
+    def test_dirichlet_tangent_field(self):
```
Comment (fritzo) [inline test/test_distributions.py]: @martinjankowiak Could you please review this test?

Comment (martinjankowiak) [reply test/test_distributions.py]: beautiful!

**Comment by fritzo** [comment]
Moving PR to pytorch/pytorch

**Pull request closed without merging**